facebook/OpenBIC
# Artemis: cb: Support to parse power brick alert event

# Description
- Support to parse power brick alert event.
- Improve function for parsing vr alert event.

# Motivation
- Support to parse power brick alert event.

# Test Plan
- Build code: Pass
- Check power brick alert event: Pass

# Log
- Main source (Q50SN120A1) root@bmc-oob:~# log-util all --print
2018 Mar 09 05:05:11 log-util: User cleared all logs
0    all      2018-03-09 05:05:30    pldmd            Device type: CB_POWER_BRICK_0 (0x3), board info: MAIN_SOURCE (0x0), event type: CML_FAULT (0x21) Assert event
0    all      2018-03-09 05:05:30    pldmd            Device type: CB_POWER_BRICK_0 (0x3), board info: MAIN_SOURCE (0x0), event type: TEMPERATURE_WARNING_FAULT (0x20) Assert event
0    all      2018-03-09 05:05:30    pldmd            Device type: CB_POWER_BRICK_0 (0x3), board info: MAIN_SOURCE (0x0), event type: UNDER_VOLTAGE (0x4) Assert event
0    all      2018-03-09 05:05:30    pldmd            Device type: CB_POWER_BRICK_0 (0x3), board info: MAIN_SOURCE (0x0), event type: OVER_CURRENT (0x3) Assert event
0    all      2018-03-09 05:05:30    pldmd            Device type: CB_POWER_BRICK_0 (0x3), board info: MAIN_SOURCE (0x0), event type: OVER_VOLTAGE (0x2) Assert event
0    all      2018-03-09 05:05:30    pldmd            Device type: CB_POWER_BRICK_1 (0x13), board info: MAIN_SOURCE (0x0), event type: CML_FAULT (0x21) Assert event
0    all      2018-03-09 05:05:30    pldmd            Device type: CB_POWER_BRICK_1 (0x13), board info: MAIN_SOURCE (0x0), event type: POWER_OFF_FAULT (0x1f) Assert event
0    all      2018-03-09 05:05:30    pldmd            Device type: CB_POWER_BRICK_1 (0x13), board info: MAIN_SOURCE (0x0), event type: POWER_GOOD_FAULT (0x1e) Assert event
0    all      2018-03-09 05:05:30    pldmd            Device type: CB_POWER_BRICK_1 (0x13), board info: MAIN_SOURCE (0x0), event type: INPUT_VOLTAGE_FAULT (0x1d) Assert event
0    all      2018-03-09 05:05:30    pldmd            Device type: CB_POWER_BRICK_1 (0x13), board info: MAIN_SOURCE (0x0), event type: OUTPUT_CURRENT_WARNING_FAULT (0x1c) Assert event
0    all      2018-03-09 05:05:30    pldmd            Device type: CB_POWER_BRICK_1 (0x13), board info: MAIN_SOURCE (0x0), event type: OUTPUT_VOLTAGE_WARNING_FAULT (0x1b) Assert event

- Second source (BMR351) root@bmc-oob:~# log-util all --print
2018 Mar 09 05:05:11 log-util: User cleared all logs
0    all      2018-03-09 04:38:43    pldmd            Device type: CB_POWER_BRICK_0 (0x3), board info: SECOND_SOURCE (0x1), event type: CML_FAULT (0x21) Assert event
0    all      2018-03-09 04:38:43    pldmd            Device type: CB_POWER_BRICK_1 (0x13), board info: SECOND_SOURCE (0x1), event type: NO_LISTED_FAULT (0x23) Assert event
0    all      2018-03-09 04:38:48    pldmd            Device type: CB_POWER_BRICK_0 (0x3), board info: SECOND_SOURCE (0x1), event type: CML_FAULT (0x21) Deassert event
0    all      2018-03-09 04:38:48    pldmd            Device type: CB_POWER_BRICK_1 (0x13), board info: SECOND_SOURCE (0x1), event type: NO_LISTED_FAULT (0x23) Deassert event
